docs/pages/developer-guide.md

```diff
@@ -936,6 +935,174 @@ In the example above, a server task can be accessed through the address:
+
+A **.truststore** is a JKS file that contains DC/OS root CA certificate stored as a trust certificate. The certificte is stored under alias **`dcos-root`**. The keystore is protected by **`notsecure`** password.
+
+The password **`notsecure`** that is protecting both JKS files (containing an end-entity certificate with private key and root CA certificate) is selected because most of Java tools and libraries validate for some password. It is not to meant to provide any additional protection. Security of both files is achieved by using DC/OS secrets store with file-based in-memory secrets. No other schedulers and tasks can access TLS artificates provisioned by a scheduler.
+
+**Imporant note:** The DC/OS 1.10 secrets store doesn't provide support for storing binary data. To overcome this limitation the `keystore` and `truststore` files are encoded using Base64 encoding when stored to secrets store. Files are mounted into a task container with additional `base64` file name suffix - `*.keystore.base64` and `*.truststore.base64`. There is a `bootstrap` utility helper that automatically decodes files in `$MESOS_SANDBOX` with `*.keystore.base64` and `*.truststore.base64` extension and stores them in a sandbox directory with expected `*.keystore` and `*.truststore` names. **Decoded files aren't stored using in-memory file system and are stored on a disk even after task finishes until the sandbox is removed by an operator or Mesos GC.**
+
+## Installation requirements
+
+To enable TLS support a `Mesosphere DC/OS Enterprise` cluster must be installed in `permissive` or `strict` security mode.
+
+A scheduler must run with `service account` that has permissions to access:
+
+- `DC/OS CA` - requires `full` permission to [`dcos:adminrouter:ops:ca:rw`](https://docs.mesosphere.com/1.9/security/perms-reference/#admin-router) resource
+
+- Secrets store - requires `full` permission [`dcos:secrets:default:<service-name>/`](https://docs.mesosphere.com/1.9/security/perms-reference/#secrets) and `full` permission on `dcos:secrets:list:default:<service-name>`
+
+  This is a minimal permission that would allow scheduler to store secrets in service namespace. However with only this permission a scheduler wouldn't be able to clean up created secrets when service is uninstalled. This is a known limitation and will be addressed in `DC/OS` version `1.11`.
+
+  Alternatively it is possible to assign service account broad `full` permission on `dcos:superuser` resource. This is a security risk so it is not adviced to run any scheduler with such a permission.
+
+  Operator can manually clean up secrets via DC/OS UI or CLI before uninstalling service that uses TLS. If operator would like to use `dcos:superuser` permission for automated uninstall it would be possible to assign such permission just before uninstalling package and removing the permission from account after the service has been uninstalled.
```

Comment:
The operate can clean up secrets manually via the DC/OS GUI or CLI before uninstalling a service... 

If the operator would like to use the `dcos:superuser` permission for automated uninstall, it would be possible to assign this permission just before uninstalling the package and remove it from account after the service has been uninstalled.